File: src/com/company/Todo.java
package com.company;

import java.util.ArrayList;

public class Todo implements Task {

    ArrayList<Step> steps = new ArrayList<Step>();
    String location = "";
    boolean isComplete = false;

    public ArrayList<Step> getSteps() {
        return steps;
    }

    public void setSteps(ArrayList<Step> steps) {
        this.steps = steps;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public boolean isComplete() {
        return isComplete;
    }

    public void setComplete(boolean complete) {
        isComplete = complete;
    }


    @Override
    public int getTaskId() {
        return 0;
    }

    @Override
    public String getTaskName() {
        return null;
    }

    @Override
    public void setTaskId() {

    }

    @Override
    public void setTaskName() {

    }
}

File: src/com/company/Todo.kt
package com.company

import java.util.ArrayList

class Todo(internal var taskId: Int, internal var taskName: String, var steps: ArrayList<Step>, var location: String) : Task {

    val isComplete: Boolean
        get() {
            for (i in steps.indices) {
                if (!steps[i].isStepComplete) {
                    return false
                }
            }
            return true
        }

    fun swapSteps(id1: Int, id2: Int) = steps.swap(id1, id2)


    override fun getTaskId(): Int {
        return taskId
    }

    override fun getTaskName(): String {
        return taskName
    }

    override fun setTaskId(taskId: Int) {
        this.taskId = taskId
    }

    override fun setTaskName(taskName: String) {
        this.taskName = taskName
    }

    fun <T> ArrayList<T>.swap(i: Int, j: Int) {
        // “this” is the list of steps
        val tmp = this[i]
        this[i] = this[j]
        this[j] = tmp
    }


}
